src/workerd/jsg/observer.h

```diff
@@ -152,10 +156,20 @@ struct InternalExceptionObserver {
   virtual void reportInternalException(const kj::Exception&, Detail detail) {}
 };
 
+WD_STRONG_BOOL(IsCodeLike);
+
 struct IsolateObserver: public CompilationObserver,
                         public InternalExceptionObserver,
                         public ResolveObserver {
   virtual ~IsolateObserver() noexcept(false) {}
+
+  // Called when eval(), new Function(), or similar dynamic code generation
+  // is performed. Note that the source here may not be a string if isCodeLike
+  // is YES.
+  virtual void onDynamicEval(
+      v8::Local<v8::Context> context, v8::Local<v8::Value> source, IsCodeLike isCodeLie) {
```

Comment:
```suggestion
  virtual void onDynamicEval(
      v8::Local<v8::Context> context, v8::Local<v8::Value> source, IsCodeLike isCodeLike) {
```

The cake is a lie, to quote a 2007 meme.